packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -0,0 +1,297 @@
+        const storage = this.DATA_MAP[name].storage
+        const key = this.DATA_MAP[name].key
+        return storage.get(key)
+    }
+
+    private set(name: AuthDataKeys, value: string, options?: any) {
+        const key = this.DATA_MAP[name].key
+        const storage = this.DATA_MAP[name].storage
+        storage.set(key, value, options)
+        this.DATA_MAP[name].callback?.()
+    }
+
+    /**
+     * A map of the data that this auth module stores. This maps the name of the property to
+     * the storage and the key when stored in that storage. You can also pass in a "callback"
+     * function to do extra operation after a property is set.
+     *
+     * @Internal
+     */
+    private get DATA_MAP(): AuthDataMap {
+        return {
+            access_token: {
+                storage: this.localStorage,
+                key: 'cc-ax',
+            },
+            customer_id: {
+                storage: this.localStorage,
+                key: 'customer_id',
+            },
+            usid: {
+                storage: this.localStorage,
+                key: 'usid',
+            },
+            enc_user_id: {
+                storage: this.localStorage,
+                key: 'enc_user_id',
+            },
+            expires_in: {
+                storage: this.localStorage,
+                key: 'expires_in',
+            },
+            id_token: {
+                storage: this.localStorage,
+                key: 'id_token',
+            },
+            idp_access_token: {
+                storage: this.localStorage,
+                key: 'idp_access_token',
+            },
+            token_type: {
+                storage: this.localStorage,
+                key: 'token_type',
+            },
+            refresh_token_guest: {
+                storage: this.cookieStorage,
+                key: 'cc-nx-g',
+                callback: () => {
+                    this.cookieStorage.delete('cc-nx')
+                },
+            },
+            refresh_token_registered: {
+                storage: this.cookieStorage,
+                key: 'cc-nx',
+                callback: () => {
+                    this.cookieStorage.delete('cc-nx-g')
+                },
+            },
+        }
+    }
+
+    private get data(): ShopperLoginTypes.TokenResponse {
+        return {
+            access_token: this.get('access_token'),
+            customer_id: this.get('customer_id'),
+            enc_user_id: this.get('enc_user_id'),
+            expires_in: parseInt(this.get('expires_in')),
+            id_token: this.get('id_token'),
+            idp_access_token: this.get('idp_access_token'),
+            refresh_token: this.get('refresh_token_registered') || this.get('refresh_token_guest'),
+            token_type: this.get('token_type'),
+            usid: this.get('usid'),
+        }
+    }
+
+    private isTokenExpired(token: string) {
+        if (!token) {
+            return true
+        }
```

Comment:
Thoughts on throwing an error in this case instead of returning a valid return value?

Reply:
I feel like, from the caller's perspective, it doesn't really matter if the token does not exists. If we throw an error here, caller would still need to catch the error and convert that to a "true"...

Reply:
I think the caller would have a condition that looks more like this `if(hasToken && !tokenExpired)` so they wouldn't even call this method if there is no token. Just seems a little weird I suppose. If I told you that isTokenExpired was true.. based on that name you should be able to assume there is a token, but thats not true with this implementation.